lib/DBDish/Pg/StatementHandle.pm6

```diff
@@ -60,6 +63,9 @@ submethod BUILD(:$!parent!, :$!pg_conn!, # Per protocol
 method execute(*@params) {
     self!enter-execute(@params.elems, @!param_type.elems);
 
+    $!parent.lock-connection();
+    LEAVE {$!parent.unlock-connection();}
```

Comment:
This should be reasonably reliable, however I think factoring this out to something like `$!parent.protect-connection: { ... }` makes mistakes less likely, especially in future refactoring.

Reply:
That does seem nicer.